plutus-tx/src/PlutusTx/Show/TH.hs

```diff
@@ -46,5 +77,194 @@ concatBuiltinStrings = \case
+    TH.ConstructorInfo
+        { TH.constructorName = conName
+        , TH.constructorVariant = TH.NormalConstructor
+        , TH.constructorFields = argTys@(_ : _)
+        } -> do
+            args <-
+                Haskell.traverse
+                    (\i -> TH.newName ("arg" <> Haskell.show i))
+                    [1 .. length argTys]
+
+            if isNonUnitTuple conName
+                then do
+                    let showArgExps :: [TH.ExpQ]
+                        showArgExps = deriveShowExpForArg 0 <$> args
+                        parenCommaArgExps =
+                            (TH.varE 'showString `TH.appE` TH.stringE "(") :
+                            intersperse (TH.varE 'showString `TH.appE` TH.stringE ",") showArgExps
+                        mappendArgs =
+                            Haskell.foldr
+                                (`TH.infixApp` TH.varE '(Haskell..))
+                                (TH.varE 'showString `TH.appE` TH.stringE ")")
+                                parenCommaArgExps
+                        pats = TH.conP conName (TH.varP <$> args)
+                        body = TH.normalB mappendArgs
+                    TH.match pats body []
+                else do
+                    let showArgExps :: [TH.ExpQ]
+                        showArgExps = deriveShowExpForArg appPrec1 <$> args
+
+                        mappendArgs, namedArgs :: TH.ExpQ
+                        mappendArgs = Haskell.foldr1 alg showArgExps
+                          where
+                            alg :: TH.ExpQ -> TH.ExpQ -> TH.ExpQ
+                            alg argExp acc = [|$argExp . $(TH.varE 'showSpace) . $acc|]
+                        namedArgs =
+                            [|
+                                $(TH.varE 'showString)
+                                    $(TH.stringE (parenInfixConName conName <> " "))
+                                    . $mappendArgs
+                                |]
+                    let pats = TH.conP conName (TH.varP <$> args)
+                        body =
+                            TH.normalB
+                                [|
+                                    $(TH.varE 'showParen)
+                                        ( $(TH.varE p)
+                                            `greaterThanInteger` $(TH.litE (TH.integerL appPrec))
+                                        )
+                                        $namedArgs
+                                    |]
+                    TH.match pats body []
+    TH.ConstructorInfo
+        { TH.constructorName = conName
+        , TH.constructorVariant = TH.RecordConstructor argNames
+        , TH.constructorFields = argTys@(_ : _)
+        } -> do
+            args <-
+                Haskell.traverse
+                    (\i -> TH.newName ("arg" <> Haskell.show i))
+                    [1 .. length argTys]
+            let showArgExps :: [TH.ExpQ]
+                -- The `dropEnd` drops the last comma
```

Comment:
why don't we just intersperse the commas like we did before?

Reply:
In the RecordConstructor case, we need to add "," after every two elements, so `intersperse` is not a good fit.